### PR TITLE
Add api key to rest service for more security

### PR DESCRIPTION
**Pull request by Arvant**
user now can run app with --key apikey to access rest service
I think it's more secure
if apikey not set it works as before
all request most have x-api-key key in headers

### CODE REVIEW

**Comment by 2bigO** [comment]
This looks great! Thanks @Arvant, I'll include this feature in a version I publish later today.

File: src/index.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ const jobsReady = agenda._ready
   });
 
 const getJobMiddleware = (jobAssertion, jobOperation, errorCode = 400) => async (ctx, next) => {
+  if(ctx.request.headers['x-api-key'] != settings.appId || settings.appId == "")
```
Comment (2bigO) [inline src/index.js]: The CI errors you're getting are probably due to this line, think the fix is to check `settings.appId` exists first.

File: settings.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ let dbhost = process.env.DB_HOST || 'localhost';
 let collection = 'agendaJobs';
 let definitions = 'jobDefinitions';
 let timeout = 5000;
-
+let appId = "";
```
Comment (2bigO) [inline settings.js]: And can have this unset initially, simply `let appId;`

File: src/index.js [hunk 3]
```diff
@@ -39,6 +41,8 @@ const getJobMiddleware = (jobAssertion, jobOperation, errorCode = 400) => async
 };
 
 const listJobs = async (ctx, next) => {
+  if(ctx.request.headers['x-api-key'] != settings.appId || settings.appId == "")
```
Comment (2bigO) [inline src/index.js]: Then, again, check here first.

**Review by 2bigO — CHANGES_REQUESTED** [review]
This can be merged as soon as CI won't complain, but looks great, thanks!

File: src/index.js (hunk 1 above)
Comment (2bigO) [reply src/index.js]: I guess `settings.appId && ctx.request.headers['x-api-key'] != settings.appId` would work.

**Comment by Arvant** [comment]
این که باز ارور داد :(

**Comment by 2bigO** [comment]
Nice job, now you are only getting lint errors, semantics are fine.

https://travis-ci.org/agenda/agenda-rest/jobs/603217247#L421

**Comment by 2bigO** [comment]
Nice job, build passed!

**Comment by Arvant** [comment]
I think I could do it :) thank you  @keyvan-m-sadeghi

**Comment by 2bigO** [comment]
Do you want to change the error code to 403 before merge?

**Comment by Arvant** [comment]
> Do you want to change the error code to 403 before merge?

changed

**Pull request merged**